Cannot create new ip range

This client code:
new_range = client.ip_ranges.create(ip_rng_attrs['start_ip'], ip_rng_attrs['end_ip'])

Is generating this error:
Traceback (most recent call last):
  File "foundation/bin/maas-config-networks", line 290, in <module>
    sys.exit(main())
  File "foundation/bin/maas-config-networks", line 285, in main
    resolve_vlans(**params)
  File "foundation/bin/maas-config-networks", line 205, in resolve_vlans
    apply_vlan_attr(attr, value)
  File "foundation/bin/maas-config-networks", line 187, in apply_vlan_attr
    end_ip=ip_rng_attrs['end_ip'])
  File "/home/ubuntu/cpe-deployments/libmaas-venv/lib/python3.5/site-packages/maas/client/utils/async.py", line 49, in wrapper
    result = eventloop.run_until_complete(result)
  File "/usr/lib/python3.5/asyncio/base_events.py", line 387, in run_until_complete
    return future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/asyncio/tasks.py", line 239, in _step
    result = coro.send(None)
  File "/home/ubuntu/cpe-deployments/libmaas-venv/lib/python3.5/site-packages/maas/client/viscera/ipranges.py", line 69, in create
    return cls._object(await cls._handler.create(**params))
  File "/home/ubuntu/cpe-deployments/libmaas-venv/lib/python3.5/site-packages/maas/client/bones/__init__.py", line 302, in __call__
    response = await self.bind(**params).call(**data)
  File "/home/ubuntu/cpe-deployments/libmaas-venv/lib/python3.5/site-packages/maas/client/bones/__init__.py", line 390, in call
    uri, body, headers = self.prepare(data)
  File "/home/ubuntu/cpe-deployments/libmaas-venv/lib/python3.5/site-packages/maas/client/bones/__init__.py", line 420, in prepare
    self.action.op, self.action.method, self.uri, data)
  File "/home/ubuntu/cpe-deployments/libmaas-venv/lib/python3.5/site-packages/maas/client/utils/__init__.py", line 111, in prepare_payload
    message = build_multipart_message(data)
  File "/home/ubuntu/cpe-deployments/libmaas-venv/lib/python3.5/site-packages/maas/client/utils/multipart.py", line 124, in build_multipart_message
    for payload in make_payloads(name, content):
  File "/home/ubuntu/cpe-deployments/libmaas-venv/lib/python3.5/site-packages/maas/client/utils/multipart.py", line 118, in make_payloads
    "%r is unrecognised: %r" % (name, content))
AssertionError: 'type' is unrecognised: <IPRangeType.RESERVED: 'reserved'>
Makefile:105: recipe for target 'networkconfig' failed
make: *** [networkconfig] Error 1

It was my understanding that 'type' is optional and not required, therefore this is happening with default value.

--- a/maas/client/viscera/ipranges.py
+++ b/maas/client/viscera/ipranges.py
@@ -50,10 +50,15 @@ class IPRangesType(ObjectType):
         :returns: The created IPRange
         :rtype: `IPRange`
         """
+        if not isinstance(type, IPRangeType):
+            raise TypeError(
+                "type must be an IPRangeType, not %s"
+                % TYPE(type).__name__)
+
         params = {
             'start_ip': start_ip,
             'end_ip': end_ip,
-            'type': type,
+            'type': type.value,
         }
         if comment is not None:
             params["comment"] = comment
@@ -73,7 +78,7 @@ class IPRanges(ObjectSet, metaclass=IPRangesType):
     """The set of IPRanges stored in MAAS."""
 
 
-class IPRangeType(ObjectType):
+class IPRangeTypeMeta(ObjectType):
     """Metaclass for `IPRange`."""
 
     async def read(cls, id: int):
@@ -82,7 +87,7 @@ class IPRangeType(ObjectType):
         return cls(data)
 
 
-class IPRange(Object, metaclass=IPRangeType):
+class IPRange(Object, metaclass=IPRangeTypeMeta):
     """A IPRange."""
 
     id = ObjectField.Checked(

--- a/maas/client/viscera/tests/test_ipranges.py
+++ b/maas/client/viscera/tests/test_ipranges.py
@@ -53,6 +53,17 @@ class TestIPRanges(TestCase):
             comment=comment,
         )
 
+    def test__ipranges_create_requires_IPRangeType(self):
+        IPRanges = make_origin().IPRanges
+        start_ip = make_string_without_spaces()
+        end_ip = make_string_without_spaces()
+        comment = make_string_without_spaces()
+        error = self.assertRaises(
+            TypeError, IPRanges.create,
+            start_ip=start_ip, end_ip=end_ip,
+            type=make_string_without_spaces(), comment=comment)
+        self.assertEquals("type must be an IPRangeType, not str", str(error))
+
     def test__ipranges_read(self):
         """IPRanges.read() returns a list of IPRanges."""
         IPRanges = make_origin().IPRanges

--- a/maas/client/viscera/tests/test_ipranges.py
+++ b/maas/client/viscera/tests/test_ipranges.py
@@ -10,6 +10,7 @@ from ..ipranges import (
 )
 
 from .. testing import bind
+from ...enum import IPRangeType
 from ...testing import (
     make_string_without_spaces,
     TestCase,
@@ -30,13 +31,13 @@ class TestIPRanges(TestCase):
         IPRanges = make_origin().IPRanges
         start_ip = make_string_without_spaces()
         end_ip = make_string_without_spaces()
-        type = make_string_without_spaces()
+        type = IPRangeType.DYNAMIC
         comment = make_string_without_spaces()
         IPRanges._handler.create.return_value = {
             "id": 1,
             "start_ip": start_ip,
             "end_ip": end_ip,
-            "type": type,
+            "type": type.value,
             "comment": comment,
         }
         IPRanges.create(
@@ -48,7 +49,7 @@ class TestIPRanges(TestCase):
         IPRanges._handler.create.assert_called_once_with(
             start_ip=start_ip,
             end_ip=end_ip,
-            type=type,
+            type=type.value,
             comment=comment,
         )
 

